system-tests/lib/cre/features/ocr/ocr.go

```diff
@@ -0,0 +1,528 @@
+			Config:          consensusV1DON.resolveOcr3Config(*ocr3Config),
+			DryRun:          false,
+		},
+	)
+	if ocr3Err != nil {
+		return nil, errors.Wrap(ocr3Err, "failed to configure OCR3 contract")
+	}
+
+	// configure DON Time contract
+	// don time happens to be the same as consensus v1 DON, but it doesn't have to be
+	_, donTimeErr := operations.ExecuteOperation(
+		creEnv.CldfEnvironment.OperationsBundle,
+		ks_contracts_op.ConfigureOCR3Op,
+		ks_contracts_op.ConfigureOCR3OpDeps{
+			Env: creEnv.CldfEnvironment,
+		},
+		ks_contracts_op.ConfigureOCR3OpInput{
+			ContractAddress: &donTimeAddr,
+			ChainSelector:   creEnv.DonTopology.HomeChainSelector,
+			DON:             consensusV1DON.keystoneDonConfig(),
+			Config:          consensusV1DON.resolveOcr3Config(*ocr3Config),
+			DryRun:          false,
+		},
+	)
+	if donTimeErr != nil {
+		return nil, errors.Wrap(donTimeErr, "failed to configure DON Time contract")
+	}
+
+	// return capabilities registry configuration data
+	capabilities := make(map[int][]keystone_changeset.DONCapabilityWithConfig)
+	for donIdx, don := range creEnv.DonTopology.Dons.List() {
+		if capabilities[donIdx] == nil {
+			capabilities[donIdx] = []keystone_changeset.DONCapabilityWithConfig{}
+		}
+		if don.HasFlag(flag) {
+			capabilities[donIdx] = append(capabilities[donIdx], keystone_changeset.DONCapabilityWithConfig{
+				Capability: kcr.CapabilitiesRegistryCapability{
+					LabelledName:   "offchain_reporting",
+					Version:        "1.0.0",
+					CapabilityType: 2, // CONSENSUS
+					ResponseType:   0, // REPORT
+				},
+				Config: &capabilitiespb.CapabilityConfig{},
+			})
+		}
+	}
+
+	return &cre.PostDONStartupOutput{
+		DONCapabilityWithConfigs: capabilities,
+	}, nil
+}
+
+func createJobs(
+	ctx context.Context,
+	creEnv *cre.Environment,
+) error {
+	ocr3Key := datastore.NewAddressRefKey(
+		creEnv.DonTopology.HomeChainSelector,
+		datastore.ContractType(keystone_changeset.OCR3Capability.String()),
+		semver.MustParse("1.0.0"),
+		contracts.OCR3ContractQualifier,
+	)
+	ocr3CapabilityAddress, err := creEnv.CldfEnvironment.DataStore.Addresses().Get(ocr3Key)
+	if err != nil {
+		return errors.Wrap(err, "failed to get OCR3 capability address")
+	}
+
+	donTimeKey := datastore.NewAddressRefKey(
+		creEnv.DonTopology.HomeChainSelector,
+		datastore.ContractType(keystone_changeset.OCR3Capability.String()),
+		semver.MustParse("1.0.0"),
+		contracts.DONTimeContractQualifier,
+	)
+	donTimeAddress, err := creEnv.CldfEnvironment.DataStore.Addresses().Get(donTimeKey)
+	if err != nil {
+		return errors.Wrap(err, "failed to get DON Time address")
+	}
+
+	jobSpecs := []*job.ProposeJobRequest{}
+
+	chainID, chErr := chainselectors.ChainIdFromSelector(creEnv.DonTopology.HomeChainSelector)
+	if chErr != nil {
+		return errors.Wrapf(chErr, "failed to get chain ID from chain selector %d", creEnv.DonTopology.HomeChainSelector)
+	}
+
+	for _, don := range creEnv.DonTopology.Dons.List() {
```

Comment:
Can we somehow scope the API to the DON? The command is called Pre/PostDON after all, seems like the fact we are needing to filter this out implies we have the boundaries wrong.

Reply:
I hear you. 

Scoping to a single DON could limit us severely.  One situation that comes to mind is generation of gateway jobspec, which requires access to all DONs, so that we can gather node addresses to be whitelisted.

Maybe we should rename this function? Because it's scope is definitely not limited to just DONs. `PostEnvStartup`?